examples/morpheusvm/actions/transfer.go

```diff
@@ -82,25 +81,33 @@ func (*Transfer) ComputeUnits(chain.Rules) uint64 {
 	return TransferComputeUnits
 }
 
-func (t *Transfer) Size() int {
-	return codec.AddressLen + consts.Uint64Len + codec.BytesLen(t.Memo)
-}
-
-func (t *Transfer) Marshal(p *codec.Packer) {
-	p.PackAddress(t.To)
-	p.PackUint64(t.Value)
-	p.PackBytes(t.Memo)
-}
-
-func UnmarshalTransfer(p *codec.Packer) (chain.Action, error) {
-	var transfer Transfer
-	p.UnpackAddress(&transfer.To) // we do not verify the typeID is valid
-	transfer.Value = p.UnpackUint64(true)
-	p.UnpackBytes(MaxMemoSize, false, &transfer.Memo)
-	return &transfer, p.Err()
-}
-
 func (*Transfer) ValidRange(chain.Rules) (int64, int64) {
 	// Returning -1, -1 means that the action is always valid.
 	return -1, -1
 }
+
+// Optional methods to override for performance optimization when hitting TPS limits
+
+// var _ chain.HasSize = (*Transfer)(nil)
+
+// func (t *Transfer) Size() int {
+// 	// For manual calculation:
+// 	return codec.AddressLen + consts.Uint64Len + codec.BytesLen(t.Memo)
+// }
+
+// var _ chain.HasMarshal = (*Transfer)(nil)
+
+// func (t *Transfer) Marshal(p *codec.Packer) {
+// 	p.PackAddress(t.To)
+// 	p.PackLong(t.Value)
+// 	p.PackBytes(t.Memo)
+// }
+
+// Register this in registry/registry.go as the second argument to Action.Register
+// func UnmarshalTransfer(p *codec.Packer) (chain.Action, error) {
+// 	var transfer Transfer
+// 	p.UnpackAddress(&transfer.To)
+// 	transfer.Value = p.UnpackUint64(true)
+// 	p.UnpackBytes(MaxMemoSize, false, &transfer.Memo)
+// 	return &transfer, p.Err()
+// }
```

Comment:
Let's uncomment these and revert this diff to a comment over the original functions to explain that these are optional to implement - we should avoid keeping commented out code in the codebase